### DISCUSSION TITLE
Extended scalar for Url implementation

### DISCUSSION THREAD

**Discussion by timpamungkas** [Netflix/dgs-framework#313] [Q&A] [answered]
Trying to add  `Url` extended scalars.
On build.gradle.

```
	implementation(platform('com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:3.12.0'))
	implementation 'com.netflix.graphql.dgs:graphql-dgs-spring-boot-starter'
	implementation 'com.netflix.graphql.dgs:graphql-dgs-extended-scalars'
```

The schema

```
scalar Date
scalar Url

type MobileApp {
    name: String!
    version: String
    platform: [String]
    releaseDate: Date
    homepage: Url
}
```

Generated class by gradle plugin.
```java
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.time.LocalDate;
import java.util.List;

public class MobileApp {
  private String name;

  private String version;

  private List<String> platform;

  private LocalDate releaseDate;

  private Url homepage;
```

The homepage data type is Url, and not `java.net.URL` as stated in [documentation](https://github.com/graphql-java/graphql-java-extended-scalars#other-scalars). Also, no import statement, so the generated class is invalid compile error.  
Is this a bug on dgs, or my implementation is wrong?

Thanks

**Comment by berngp** [comment, marked as answer]
The codegen plugin requires an explicit [Type Mapping](https://netflix.github.io/dgs/generating-code-from-schema/). This caveat is missing in the [extended scalar docs](https://netflix.github.io/dgs/scalars/), will add it tomorrow.

**Reply by timpamungkas** [reply]
Added this

```
generateJava {
    typeMapping = ["Url"           : "java.net.URL",
                   "Time"          : "java.time.OffsetTime",
                   "DateTime"      : "java.time.OffsetDateTime",
                   "Date"          : "java.time.LocalDate",
                   "JSON"          : "com.fasterxml.jackson.databind.JsonNode",
                   "PositiveInt"   : "java.lang.Integer",
                   "NegativeFloat" : "java.lang.Double",
                   "NonPositiveInt": "java.lang.Integer"]
}
```

Thanks

**Reply by berngp** [reply]
The Spring Boot _auto-configuration_ available in  `com.graphql-java:graphql-java-extended-scalars` doesn't affect the _CodeGen Plugin_, its meant to be used by the server such that it understand how to SerDe types in the GraphQL request.

**Reply by timpamungkas** [reply]
Submitted [PR](https://github.com/Netflix/dgs/pull/38) for docs, thanks

**Reply by berngp** [reply]
Thanks for the PR, I reworded it a bit and merged. Result:

![image](https://user-images.githubusercontent.com/134985/117524415-016d3980-af72-11eb-95d9-77dbfc81651a.png)

**Reply by berngp** [reply]
Just noticed you are using `JsonNode` to map it to the `JSON` scalar. Not sure that will work, probably you want to map `JSON` to just a `"java.util.Map"`.
